test/internal_rules.jl

```diff
@@ -386,5 +387,39 @@ end
         @test isapprox(dA, dA_sym)
     end
 end
+
+@testset "Linear solve for triangular matrices" begin
+    @testset for T in (UpperTriangular, LowerTriangular, UnitUpperTriangular, UnitLowerTriangular),
+        TE in (Float64, ComplexF64), sizeB in ((3,), (3, 3))
+        M = rand(TE, 3, 3)
+        B = rand(TE, sizeB...)
+        Y = zeros(TE, sizeB...)
+        A = T(M)
+        @testset "test primal" begin
+            Y1 = deepcopy(Y)
+            Y2 = deepcopy(Y)
+            dY = make_zero(Y)
+            function f!(Y, A, B)
+                ldiv!(Y, A, B)
+                return nothing
+            end
+            ldiv!(Y1, A, B)
+            Enzyme.autodiff(Reverse, f!, Duplicated(Y2, dY), Const(A), Const(B))
+            @test Y1 ≈ Y2
+        end
+        @testset "test against EnzymeTestUtils through constructor" begin
+            _A = T(A)
+            function f!(Y, A, B, ::T) where T
+                ldiv!(Y, T(A), B)
+                return nothing
+            end
+            for TY in (Const, Duplicated, BatchDuplicated),
+                TM in (Const, Duplicated, BatchDuplicated),
+                TB in (Const, Duplicated, BatchDuplicated)
+                test_reverse(f!, Const, (Y, TY), (M, TM), (B, TB), (_A, Const))
```

Comment:
This should allow the tests to pass. Only certain combinations of activities are valid (e.g. can't mix `Duplicated` and `BatchDuplicated`), and the invalid combinations are the ones causing the current failures:
```suggestion
                are_activities_compatible(Const, TY, TA, TB) || continue
                test_reverse(f!, Const, (Y, TY), (M, TM), (B, TB), (_A, Const))
```